tools/licenses/lib/licenses.dart

```diff
@@ -116,21 +153,132 @@ LicenseType convertBodyToType(String body) {
+    result.writeln('=' * 100);
+    (libraries.map((String s) => 'LIBRARY: $s').toList()..sort()).forEach(result.writeln);
+    (origins.map((String s) => 'ORIGIN: $s').toList()..sort()).forEach(result.writeln);
+    result.writeln('TYPE: $type');
+    (targets.map((String s) => 'FILE: $s').toList()..sort()).forEach(result.writeln);
+    result.writeln('-' * 100);
+    if (body.isEmpty) {
+      result.writeln('<THIS BLOCK INTENTIONALLY LEFT BLANK>');
+    } else {
+      result.writeln(body);
+    }
+    result.writeln('=' * 100);
+    return result.toString();
+  }
+
+  String toStringFormal() {
+    final StringBuffer result = StringBuffer();
+    (libraries.toList()..sort()).forEach(result.writeln);
+    result.writeln();
+    assert(body.isNotEmpty);
+    result.write(body);
+    return result.toString();
+  }
+}
+
+/*
+// TODO(ianh): Remove this after the PR lands, to reorder the licenses to the new order.
+String _oldProcess(String body) {
+  if (body == '<THIS BLOCK INTENTIONALLY LEFT BLANK>') {
+    return body;
+  }
+  final List<String> lines = body.split('\n');
+  while (lines.isNotEmpty && lines.first == '') {
+    lines.removeAt(0);
+  }
+  while (lines.isNotEmpty && lines.last == '') {
+    lines.removeLast();
+  }
+  if (lines.isEmpty) {
+    return '';
+  }
+  final List<String?> output = <String?>[];
+  int? lastGood;
+  String? previousPrefix;
+  bool lastWasEmpty = true;
+  for (final String line in lines) {
+    final Match match = stripDecorations.firstMatch(line)!;
+    final String? prefix = match.group(1);
+    String? s = match.group(2);
+    if (!lastWasEmpty || s != '') {
+      if (s != '') {
+        if (previousPrefix != null) {
+          if (previousPrefix.length > prefix!.length) {
+            previousPrefix = prefix;
+          } else if (previousPrefix.length < prefix.length) {
+            s = '${prefix.substring(previousPrefix.length)}$s';
+          }
+        } else {
+          previousPrefix = prefix;
+        }
+        lastWasEmpty = false;
+        lastGood = output.length + 1;
+      } else {
+        lastWasEmpty = true;
+      }
+      output.add(s);
+    }
+  }
+  if (lastGood == null) {
+    return '';
+  }
+  return output.take(lastGood).join('\n');
+}
+*/
+
+List<GroupedLicense> groupLicenses(Iterable<Assignment> assignments) {
+  final Map<String, GroupedLicense> groups = <String, GroupedLicense>{};
+  for (final Assignment assignment in assignments) {
+    final String body = assignment.license.toStringBody(assignment.source);
+    final GroupedLicense entry = groups.putIfAbsent(body, () => GroupedLicense(assignment.license.type, body));
+    entry.targets.add(assignment.target);
+    entry.libraries.add(assignment.source.libraryName);
+    entry.origins.add(assignment.license.origin);
+  }
+  final List<GroupedLicense> results = groups.values.toList();
+  // results.sort((GroupedLicense a, GroupedLicense b) => _oldProcess(a.body).compareTo(_oldProcess(b.body)));
```

Comment:
(and this?)

Reply:
removed. This was something I was using to help audit the changes during development but it stopped being practical once the changes got too big.